playbooks/roles/grafana/tasks/main.yml

```diff
@@ -0,0 +1,133 @@
+---
+- name: Include variables
+  include_vars: "{{ lookup('first_found', params) }}"
+  vars:
+    params:
+      files: "{{ distro_lookup_path }}"
+      paths:
+        - "vars"
+
+- name: Install required packages
+  become: true
+  ansible.builtin.package:
+    state: present
+    name: "{{ item }}"
+  loop:
+    - "{{ packages }}"
+  when: "ansible_facts.pkg_mgr != 'atomic_container'"
+  register: task_result
+  until: task_result is success
+  retries: 5
+
+- name: Create Grafana group
+  become: true
+  ansible.builtin.group:
+    name: "{{ grafana_os_user }}"
+    state: present
+
+- name: Create Grafana user for docker
+  become: true
+  ansible.builtin.user:
+    name: "{{ grafana_os_user }}"
+    group: "{{ grafana_os_group }}"
+    groups: docker 
+    state: present
+  when: container_command == "docker"
+
+- name: Create alerta user for podman
+  become: true
+  ansible.builtin.user:
+    name: "{{ grafana_os_user }}"
+    group: "{{ grafana_os_group }}"
+    state: present
+  when: container_command == "podman"
+
+- name: Create required directories
+  become: true
+  file:
+    path: "{{ item.dest }}"
+    state: directory
+    recurse: yes
+    owner: "{{ grafana_os_user }}"
+    group: "{{ grafana_os_group }}"
+    mode: "{{ item.mode | default(omit) }}"
+  loop:
+    - {dest: "/etc/grafana"}
+    - {dest: "/etc/grafana/provisioning"}
+    - {dest: "/etc/grafana/ldap"}
+    - {dest: "/var/lib/grafana", mode: "0777"}
```

Comment:
let's skip this one at all and use docker/podman volumes instead

Reply:
the first three directories are required as they hold the config we normally mount to the container. the /var/lib/grafana directory holds the installed plugins...if I use --tmpfs, then the plugins will disappear after a 'hard' reboot of the container...

Reply:
that's why I suggest using regular container volume, which survives reboot

Reply:
ok, I see